OtripleS.Web.Api.Tests.Unit/Services/Calendars/CalendarServiceTests.Validations.Delete.cs

```diff
@@ -0,0 +1,98 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Moq;
+using OtripleS.Web.Api.Models.Calendars;
+using OtripleS.Web.Api.Models.Calendars.Exceptions;
+using Xunit;
+
+namespace OtripleS.Web.Api.Tests.Unit.Services.Calendars
+{
+    public partial class CalendarServiceTests
+	{
+		[Fact]
+		public async Task ShouldThrowValidatonExceptionOnDeleteWhenIdIsInvalidAndLogItAsync()
+		{
+			// given
+			Guid randomCalendarId = default;
+			Guid inputCalendarId = randomCalendarId;
+
+			var invalidCalendarInputException = new InvalidCalendarInputException(
+				parameterName: nameof(Calendar.Id),
+				parameterValue: inputCalendarId);
+
+			var expectedCalendarValidationException =
+				new CalendarValidationException(invalidCalendarInputException);
+
+			// when
+			ValueTask<Calendar> actualCalendarTask =
+				this.calendarService.DeleteCalendarByIdAsync(inputCalendarId);
+
+			// then
+			await Assert.ThrowsAsync<CalendarValidationException>(() => actualCalendarTask.AsTask());
+
+			this.loggingBrokerMock.Verify(broker =>
+				broker.LogError(It.Is(SameExceptionAs(expectedCalendarValidationException))),
+					Times.Once);
+
+			this.storageBrokerMock.Verify(broker =>
+				broker.SelectCalendarByIdAsync(It.IsAny<Guid>()),
+					Times.Never);
+
+			this.storageBrokerMock.Verify(broker =>
+				broker.DeleteCalendarAsync(It.IsAny<Calendar>()),
+					Times.Never);
+
+			this.storageBrokerMock.VerifyNoOtherCalls();
+			this.loggingBrokerMock.VerifyNoOtherCalls();
```

Comment:
In the `VerifyNoOtherCalls` make sure you bring first the dependencies that has been called first, from a chronological order, then the ones that don't get called at all by the likelihood of when and where they would be called with respect to each other in the happy case scenario.